src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -242,48 +242,48 @@ private static void HandleLastWin32Error()
 
     internal static class NtProcessInfoHelper
     {
-        // Cache a single buffer for use in GetProcessInfos().
-        private static long[]? CachedBuffer;
-
         // Use a smaller buffer size on debug to ensure we hit the retry path.
 #if DEBUG
-        private const int DefaultCachedBufferSize = 1024;
+        private const int DefaultCachedBufferSize = 8 * 1024;
 #else
-        private const int DefaultCachedBufferSize = 128 * 1024;
+        private const int DefaultCachedBufferSize = 8 * 128 * 1024;
```

Comment:
Would it make sense to add a comment that explains why the `8` factor is necessary? Perhaps replace with sizeof(long) while you're at it?

Reply:
I've simplified it to `1024 * 1024`. Is that enough?